packages/jsxstyle-utils/src/getStyleCache.ts

```diff
@@ -2,6 +2,20 @@ import getStyleKeysForProps from './getStyleKeysForProps';
 import addStyleToHead from './addStyleToHead';
 import stringHash from './stringHash';
 
+export interface InsertRuleCallback {
+  (rule: string, props?: {}): boolean | void;
```

Comment:
I feel like you could get fancy here and type `props` more tightly, the same way that `document.createElement` returns a different type based on which string you pass it.  I think it's something like:


```typescript
export interface InsertRuleCallback<K extends keyof RuleToProps> {
  (rule: K, props?: RuleToProps[K]): boolean | void;
}
```

and it looks like this is just a function type.  I'd have expected

```ts
export type InsertRuleCallback<K extends keyof RuleToProps> = (rule: K, props?: RuleToProps[K]) => boolean | void;
```
but that's just a style thing.  I've never seen `interface` used to represent a single function signature before.

Reply:
Agree that this should probably be:
```ts
function InsertRuleCallback(rule: string, props?: {}): boolean | void;
```

Reply:
@steadicat does this look right?

```typescript
type InsertRuleCallback = (rule: string, props?: {}) => boolean | void;
```

@appsforartists for now, in order to get this PR landed, I’m going to leave this as the not-so-great generic props type, but I’d love a PR to improve this if you have the time. I’m reaching the limits of what I can google regarding TypeScript 😅